src/Aspire.Dashboard/Telemetry/DashboardTelemetrySender.cs

```diff
@@ -0,0 +1,202 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Net;
+using System.Net.Security;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading.Channels;
+using Aspire.Dashboard.Configuration;
+using Microsoft.Extensions.Options;
+
+namespace Aspire.Dashboard.Telemetry;
+
+public class DashboardTelemetrySender : IDashboardTelemetrySender
+{
+    private readonly IOptions<DashboardOptions> _options;
+    private readonly ILogger<DashboardTelemetrySender> _logger;
+    private readonly Channel<(OperationContext, Func<HttpClient, Func<OperationContextProperty, object>, Task>)> _channel;
+    private HttpClient? _httpClient;
+    private bool? _isEnabled;
+    private Task? _sendLoopTask;
+
+    // Internal for testing.
+    internal Func<HttpClientHandler, HttpMessageHandler>? CreateHandler { get; set; }
+
+    public DashboardTelemetrySender(IOptions<DashboardOptions> options, ILogger<DashboardTelemetrySender> logger)
+    {
+        _options = options;
+        _logger = logger;
+        var channelOptions = new BoundedChannelOptions(1000)
+        {
+            FullMode = BoundedChannelFullMode.DropOldest,
+            SingleReader = true
+        };
+        _channel = Channel.CreateBounded<(OperationContext, Func<HttpClient, Func<OperationContextProperty, object>, Task>)>(channelOptions);
+    }
+
+    private static bool HasDebugSession(
+        DebugSession debugSession,
+        [NotNullWhen(true)] out Uri? debugSessionUri,
+        [NotNullWhen(true)] out string? token,
+        [NotNullWhen(true)] out X509Certificate2? serverCert)
+    {
+        if (debugSession.Address is not null && debugSession.Token is not null && debugSession.GetServerCertificate() is { } cert && debugSession.TelemetryOptOut is not true)
+        {
+            debugSessionUri = new Uri($"https://{debugSession.Address}");
```

Comment:
From reading the doc here - https://github.com/dotnet/aspire/blob/main/docs/specs/IDE-execution.md#enabling-ide-execution - it seems like https is for when `DEBUG_SESSION_SERVER_CERTIFICATE` is present, otherwise it is http.

Can you update this to match the spec? Also write a unit test that checks the address specified with requests is correct depending on whether `DEBUG_SESSION_SERVER_CERTIFICATE` is set or not.